### ISSUE TITLE
[In Review] SDL 0150 - Enhancing onHMIStatus with a New Parameter for Video Streaming State

### ISSUE THREAD

**Issue by theresalech** [smartdevicelink/sdl_evolution#429]
Hello SDL community,

The review of "Enhancing onHMIStatus with a New Parameter for Video Streaming State" begins now and runs through March 27, 2018. The proposal is available here:

https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0150-video-streaming-state.md

Reviews are an important part of the SDL evolution process. All reviews should be sent to the associated Github issue at:

https://github.com/smartdevicelink/sdl_evolution/issues/430

What goes into a review?

The goal of the review process is to improve the proposal under review through constructive criticism and, eventually, determine the direction of SDL. When writing your review, here are some questions you might want to answer in your review:

* Is the problem being addressed significant enough to warrant a change to SDL?
* Does this proposal fit well with the feel and direction of SDL?
* If you have used competitors with a similar feature, how do you feel that this proposal compares to those?
* How much effort did you put into your review? A glance, a quick reading, or an in-depth study?
Please state explicitly whether you believe that the proposal should be accepted into SDL.

More information about the SDL evolution process is available at

https://github.com/smartdevicelink/sdl_evolution/blob/master/process.md

Thank you,
Theresa Lech

Program Manager - Livio
theresa@livio.io

**Comment by LuxoftAKutsan** [comment]
Propose to add additional description related to video/audio streaming: 

Projection Application may not have any other HMI types to support video/audio streaming.
isMediaApplication flag in RegisterAppInterface controls if application is able to be AUDIBLE.  
if AppHMIType=PROJECTION and isMediaApplication=true, then it will follow the HMI rules for a regular media application
if AppHMIType=PROJECTION and isMediaApplication=false, then it will follow the HMI rules of a non-media application

**Comment by LuxoftAKutsan** [comment]
Propose to include following points in proposal: 

Projection application may not have any other HMI types to support video/audio streaming.
isMediaApplication boolean parameter in register app interface should control if application support AUDIBLE streaming state.  
if AppHMIType=PROJECTION and isMediaApplication=true, then it will follow the HMI rules for a regular media application
if AppHMIType=PROJECTION and isMediaApplication=false, then it will follow the HMI rules of a non-media application

**Comment by AndrewRMitchell** [comment]
If AppHMIType=PROJECTION and isMediaApplication=false, doesn't that at the moment mean Navigation? Aren't Media and Navigation the only types of Apps where projection is supported?

Just curious.

**Comment by yang1070** [comment]
@LuxoftAKutsan I agree that 
```Projection application may not have any other HMI types to support video/audio streaming.```
Currently, besides Projection, only Navigation apps support video and audio streaming. So this also means an application cannot have AppHMIType=PROJECTION and AppHMIType=NAVIGATION at the same time if there is no new AppHMITypes defined (to support video/audio).

**Comment by joeljfischer** [comment]
@AndrewRMitchell to build on @yang1070, there is the `PROJECTION` appHMIType and the `NAVIGATION` appHMIType.

**Comment by LuxoftAKutsan** [comment]
Propose to add also formalizing of audiostreaming state rules : 
```
Case#     Current Audio source     SDL App2 comes to FULL/LIMITED             Results                         SDL Core Actions regarding audiostreamingState
1         NONE                     App2 is not media                          no app is the audion source     send onHMIStatus to app2 with NOT_AUDIBLE
2         NONE                     App2 is Media                              App2 is the audio source        send onHMIStatus to app2 with AUDIBLE
3         App1                     App2 is not media                          App1 is still the audio source  send onHMIStatus to app2 with NOT_AUDIBLE
4         App1                     App2 is Media                              App2 is the new audio source    send onHMIStatus to app1 with NOT_AUDIBLE & app2 with AUDIBLE
5         App1                     CarPlay/Android Auto is the active screen  no SDL app is the audio source  send onHMIStatus to app1 with NOT_AUDIBLE

```

**Comment by LuxoftAKutsan** [comment]
In case if applications is NAVIGATION,  isMediaApplication flag has no effect, so that is rules : 
```
Case#     Current Audio source     SDL App2 comes to FULL/LIMITED             Results                         SDL Core Actions regarding audiostreamingState
2         NONE                     App2 is Media                              App2 is the audio source        send onHMIStatus to app2 with AUDIBLE
4         App1                     App2 is Media                              App2 is the new audio source    send onHMIStatus to app1 with NOT_AUDIBLE & app2 with AUDIBLE
5         App1                     CarPlay/Android Auto is the active screen  no SDL app is the audio source  send onHMIStatus to app1 with NOT_AUDIBLE
```

**Comment by yang1070** [comment]
![audio_streaming_state](https://user-images.githubusercontent.com/12255391/37988670-8372516e-31cf-11e8-8cd0-ca9a1d880c09.png)

**Comment by yang1070** [comment]
![audio_streaming_state](https://user-images.githubusercontent.com/12255391/37988695-929c7a16-31cf-11e8-9103-6be7be8d32c9.png)

**Comment by JackLivio** [comment]
@yang1070 

```
This proposal also adds a new enum value PROJECTION to data type EventTypes. HMI shall send an OnEventChanged notification with EventTypes=PROJECTION to SDL Core when HMI brings a projection app to foreground (isActive=true) or HMI switches from the foreground projection app (isActive=false).
```

I see there is no event type for a NAVIGATION app streaming. Why is this field required for PROJECTION, and not for NAVIGATION?

**Comment by yang1070** [comment]
@JackLivio You are correct. I mixed NAVIGATION and EMBEDDED_NAVI. We shall have a separate enum value for NAVIGATION  app.

**Comment by theresalech** [comment]
The Steering Committee voted to return this proposal for revisions, per the author's request, to allow the author to update the proposal to incorporate the changes as outlined in his comments, specifically https://github.com/smartdevicelink/sdl_evolution/issues/429#issuecomment-376637792.

**Comment by theresalech** [comment]
The author has updated this proposal per the requested revisions, and those revisions are now ready for Steering Committee review.  This revised proposal will be in review until April 10, 2018.

**Comment by joeljfischer** [comment]
This looks good, I would just note that significant work may need to be done in the iOS library's `StreamingMediaManager` to support this new parameter and the old flow as well.

**Comment by theresalech** [comment]
Issues Entered:

- [RPC](https://github.com/smartdevicelink/rpc_spec/issues/73)
- [Android](https://github.com/smartdevicelink/sdl_android/issues/734)
- [iOS](https://github.com/smartdevicelink/sdl_ios/issues/929)
- [Core](https://github.com/smartdevicelink/sdl_core/issues/2129)
